test/jasmine/tests/animate_test.js

```diff
@@ -836,4 +837,43 @@ describe('animating scatter traces', function() {
             expect(gd.calcdata[0][0].y).toEqual(3);
         }).catch(failTest).then(done);
     });
+
+    it('should animate axis ranges using the less number of steps', function(done) {
+        Plotly.plot(gd, [{
+            y: [1, 2, 1]
+        }, {
+            type: 'bar',
+            y: [2, 1, 2]
+        }])
+        .then(function() {
+            spyOn(gd._fullData[0]._module.basePlotModule, 'transitionAxes').and.callThrough();
+            spyOn(gd._fullData[0]._module, 'plot').and.callThrough();
+            spyOn(gd._fullData[1]._module, 'plot').and.callThrough();
+            spyOn(Registry, 'call').and.callThrough();
+
+            return Plotly.animate('graph', {
+                layout: {
+                    xaxis: {range: [0.45, 0.55]},
+                    yaxis: {range: [0.45, 0.55]}
+                }
+            }, {
+                transition: {duration: 500},
+                frame: {redraw: false}
+            });
+        })
+        .then(function() {
+            // the only redraw should occur during Cartesian.transitionAxes,
+            // where Registry.call('relayout') is called leading to a _module.plot call
+            expect(gd._fullData[0]._module.basePlotModule.transitionAxes).toHaveBeenCalledTimes(1);
+            expect(gd._fullData[0]._module.plot).toHaveBeenCalledTimes(1);
+            expect(gd._fullData[1]._module.plot).toHaveBeenCalledTimes(1);
+            expect(Registry.call).toHaveBeenCalledTimes(1);
+
+            var calls = Registry.call.calls.all();
+            expect(calls.length).toBe(1, 'just one Registry.call call');
+            expect(calls[0].args[0]).toBe('relayout', 'called Registry.call with');
```

Comment:
This is a pretty low-level assertion, not clear to me that it would catch breaks other than the exact change you made here. Would it be possible to do something like:

```js
    var animatePromise = Plotly.animate(...)
    expect(stuff).toBe(nearWhereItStarted)
    return animatePromise;
})
.then(function() {
    expect(stuff).toBe(nearWhereItFinishes);
    expect(theFullDataAndRegistryCallStuff);
```
ie something about the scatter marker and bar positions being updated async that would have failed before this change?

Reply:
> ie something about the scatter marker and bar positions being updated async that would have failed before this change? 

The marker and bar position does change, it's the subplot and clip rect translate transform that change during axis range transitions via `Cartesian.transitionAxes` (which mimics the cartesian dragbox logic).

Better test added in -> 19300f9